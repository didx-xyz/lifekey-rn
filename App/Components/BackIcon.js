import React from "react"
import Svg, { Polyline } from "react-native-svg"

const BackIcon = (props) => {
  const dynamic = {
    "stroke": props.stroke || "#000"
  }

  return (
    <Svg width={props.width} height={props.height} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 9.4 17.7">
      <Polyline {...styles.cls1} {...dynamic} points="8.9,0.5 0.5,9 8.7,17.2"/>
    </Svg>
  )
}

const styles = {
  "cls1": {
    "fill": "none",
    "stroke": "#000",
    "strokeLinecap": "round",
    "strokeLinejoin": "round"
  }
}

export default BackIcon
