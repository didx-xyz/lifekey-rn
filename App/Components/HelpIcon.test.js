// external dependencies
import React from "react"

// internal dependencies
import BackIcon from "./BackIcon"

test("BackIcon renders correctly, by default", () => {
  const tree = shallow(
    <BackIcon width={16} height={16} />
  )

  expect(tree).toMatchSnapshot()
})

test("BackIcon renders correctly, when modified", () => {
  const tree = shallow(
    <BackIcon width={32} height={32} stroke="#fff" />
  )

  expect(tree).toMatchSnapshot()
})
