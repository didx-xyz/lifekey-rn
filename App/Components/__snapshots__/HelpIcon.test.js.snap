// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIcon renders correctly, by default 1`] = `
<Svg
  height={16}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 9.4 17.7"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <Polyline
    fill="none"
    points="8.9,0.5 0.5,9 8.7,17.2"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</Svg>
`;

exports[`BackIcon renders correctly, when modified 1`] = `
<Svg
  height={32}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 9.4 17.7"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <Polyline
    fill="none"
    points="8.9,0.5 0.5,9 8.7,17.2"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</Svg>
`;

exports[`HelpIcon renders correctly, by default 1`] = `
<Svg
  height={16}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 24.2 24.2"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <Circle
    cx="12.1"
    cy="12.1"
    fill="none"
    r="11.6"
    stroke="#000"
    strokeMiterlimit={10}
  />
  <Line
    fill="none"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
    x1="9.9"
    x2="14"
    y1="18.1"
    y2="18.1"
  />
  <Line
    fill="none"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    x1="12.1"
    x2="12.1"
    y1="6.4"
    y2="6.4"
  />
  <Polyline
    fill="none"
    points="12.4,18.1 12.4,9.7 10.2,9.7"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</Svg>
`;

exports[`HelpIcon renders correctly, when modified 1`] = `
<Svg
  height={32}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 24.2 24.2"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <Circle
    cx="12.1"
    cy="12.1"
    fill="none"
    r="11.6"
    stroke="#fff"
    strokeMiterlimit={10}
  />
  <Line
    fill="none"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
    x1="9.9"
    x2="14"
    y1="18.1"
    y2="18.1"
  />
  <Line
    fill="none"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    x1="12.1"
    x2="12.1"
    y1="6.4"
    y2="6.4"
  />
  <Polyline
    fill="none"
    points="12.4,18.1 12.4,9.7 10.2,9.7"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</Svg>
`;
